### ISSUE TITLE
[Feature Request]: 🔗 Implement blockchain integration for immutability.

### ISSUE THREAD

**Issue by vigneshs-dev** [vigneshs-dev/Q-Vote#3]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

As we have to make our Voting system more Secure even for the future, We are using Quantum entanglement. Along with that if Blockchain implemented, it will add more security to it.

### Record

- [X] I agree to follow this project's Code of Conduct
- [ ] I want to work on this issue
- [ ] I'm willing to provide further clarification or assistance if needed.

**Comment by hrit2773** [comment]
@vigneshs-dev can you provide more info about which kind of data you want to put it on blockchain, because the quantum logic is also implemented so what is required? Just to store the data or should we include any new logic in the smart contract.

**Comment by vigneshs-dev** [comment]
> @vigneshs-dev can you provide more info about which kind of data you want to put it on blockchain, because the quantum logic is also implemented so what is required? Just to store the data or should we include any new logic in the smart contract.

Hi @hrit2773 ,
Block chain is the latest most implemented technology on Voting app. The Quantum Logic is most advanced. So, I am planning to give both the security in my Q-Vote app. The block chain logic will acts as a extra output check method for Quantum Logic. Because, as of now, the Quantum Logic is prone to Noise as Quantum works on Superposition. 

I think I cleared your doubt. Feel free to ask if you still have other doubts.

**Comment by ShivRaiGithub** [comment]
Hey @vigneshs-dev  I can work on it since I have experience on blockchain.
I have this in mind : Blockchain contract creation, testing and deployment can be done through Foundry, which is a solidity framework faster than hardhat. web3.py can be used to connect frontend to contracts.
I have seen the program on the deployed site, please tell me the exact part which you want to be on the blockchain i.e. do you want the votes to be submitted only once when we "Find the winner" etc. Thank you.

**Comment by vigneshs-dev** [comment]
> Hey @vigneshs-dev I can work on it since I have experience on blockchain. I have this in mind : Blockchain contract creation, testing and deployment can be done through Foundry, which is a solidity framework faster than hardhat. web3.py can be used to connect frontend to contracts. I have seen the program on the deployed site, please tell me the exact part which you want to be on the blockchain i.e. do you want the votes to be submitted only once when we "Find the winner" etc. Thank you.

Hi @ShivRaiGithub ,

You can work on your idea. Now the votes and calculated automatically. But, I have updated the architecture to get votes from user after logging in and cast their votes. After submitting their votes, they are gonna see the results.

https://github.com/vigneshs-dev/Q-Vote/tree/login_voting_result_db

You can pull this branch and run in your system to get an idea of what I am talking. I don't know more about blockchain. As you have knowledge on this, Create your idea on implementing and share it with me. 

After running the above code, get website access through 
http://127.0.0.1:5000/register

**Comment by ShivRaiGithub** [comment]
@vigneshs-dev  Saw the files. Now, the whole project has parts that can be replaced by that of blockchain like login functionality, etc but since this is mainly based on quantum computing, I would like to keep the blockchain part to only the bare necessary.
Here is what I suggest : only the votes being casted by the user use blockchain as an added layer of security ( they will retain their original mechanism though). It will still provide security as any user can see the transactions made to the contract through a block-explorer ( a website/tool to see details of contracts and transactions ).

**Comment by vigneshs-dev** [comment]
> @vigneshs-dev Saw the files. Now, the whole project has parts that can be replaced by that of blockchain like login functionality, etc but since this is mainly based on quantum computing, I would like to keep the blockchain part to only the bare necessary. Here is what I suggest : only the votes being casted by the user use blockchain as an added layer of security ( they will retain their original mechanism though). It will still provide security as any user can see the transactions made to the contract through a block-explorer ( a website/tool to see details of contracts and transactions ).

Yeah. It would be great. Now I am updating the Quantum logic also. As there is noise in Quantum logic, I am getting wrong results. I have to solve that. If you want to build everything in block chain, You can make. As you said , we can implement that as a counter check for correct output. If you want to work on this please notify me. Thankyou for your valuable comment.

**Comment by ShivRaiGithub** [comment]
@vigneshs-dev  Since you are also modifying the Quantum logic, I have this proposal : Right now I make the contracts along documentation of how to work with them while you take your time correcting the Quantum logic. We can see the integration after that. So, is it fine if I open a new issue of "Contract creation" to be assigned to ? It will be better as it will be more specific and easier to track.

**Comment by vigneshs-dev** [comment]
> @vigneshs-dev Since you are also modifying the Quantum logic, I have this proposal : Right now I make the contracts along documentation of how to work with them while you take your time correcting the Quantum logic. We can see the integration after that. So, is it fine if I open a new issue of "Contract creation" to be assigned to ? It will be better as it will be more specific and easier to track.

Yeah. You can create separate issue for that. I will assign that issue to you. As you said, it will be more clear for future updates. 

https://github.com/vigneshs-dev/Q-Vote/tree/login_voting_result_db

This is the latest logic I am working. SignUp/Login page. Getting votes from users. You may already know that. Even though I am specifying. I am really happy to use your help on this logic part.

**Comment by ShivRaiGithub** [comment]
@vigneshs-dev  Happy to collaborate with you.
